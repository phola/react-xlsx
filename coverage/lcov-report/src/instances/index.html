<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/instances</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> src/instances
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>20/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>20/25</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="86.96" class="pic high"><div class="chart"><div class="cover-fill" style="width: 86%;"></div><div class="cover-empty" style="width:14%;"></div></div></td>
	<td data-value="86.96" class="pct high">86.96%</td>
	<td data-value="23" class="abs high">20/23</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="8" class="abs medium">6/8</td>
	<td data-value="86.96" class="pct high">86.96%</td>
	<td data-value="23" class="abs high">20/23</td>
	</tr>

<tr>
	<td class="file low" data-value="subtest.js"><a href="subtest.js.html">subtest.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Fri Sep 16 2016 08:43:05 GMT+0100 (GMT Daylight Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
