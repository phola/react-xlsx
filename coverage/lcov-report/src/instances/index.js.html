<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/instances/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/instances</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">88% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>22/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">77.78% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>7/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>22/25</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">58x</span>
<span class="cline-any cline-yes">58x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">90x</span>
<span class="cline-any cline-yes">900x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { WorkBook, Sheet, Cell, XTrigger } from '../index'
import SubTest from './subtest'
import FileSaver from 'file-saver'
&nbsp;
const defaultCellStyle = { font: { name: 'Verdana', sz: 11, color: 'FF00FF88' }, fill: { fgColor: { rgb: 'FFFFAA00' } } }
&nbsp;
const handleXLSX = <span class="fstat-no" title="function not covered" >(b</span>lob) =&gt; {
<span class="cstat-no" title="statement not covered" >  console.log(blob)</span>
<span class="cstat-no" title="statement not covered" >  FileSaver.saveAs(blob, 'test.xlsx')</span>
}
&nbsp;
export const simple = () =&gt; {
  return (&lt;WorkBook defaultCellStyle={defaultCellStyle} toXLSXCallback={handleXLSX}&gt;
            &lt;Sheet name='woooo'&gt;
              &lt;Cell
                row={0}
                col={1}
                colSpan={2}
                rowSpan={6}&gt;
                &lt;div&gt;
                  &lt;div&gt;
                    nested
                  &lt;/div&gt;
                &lt;/div&gt;
              &lt;/Cell&gt;
            &lt;/Sheet&gt;
            &lt;Sheet name='hoooo'&gt;
              &lt;Cell row={1} col={2} cellStyle={{font: {bold: true}}}&gt;
                I am bored
              &lt;/Cell&gt;
              &lt;Cell cellRef='C9'&gt;
                plain string
              &lt;/Cell&gt;
            &lt;/Sheet&gt;
            &lt;XTrigger action='onClick'&gt;
              &lt;button&gt;
                get XLSX
              &lt;/button&gt;
            &lt;/XTrigger&gt;
          &lt;/WorkBook&gt;)
}
&nbsp;
export const simpleNoCallback = () =&gt; {
  return (&lt;WorkBook defaultCellStyle={defaultCellStyle}&gt;
            &lt;Sheet name='woooo'&gt;
              &lt;Cell
                row={0}
                col={1}
                colSpan={2}
                rowSpan={6}&gt;
                &lt;div&gt;
                  &lt;div&gt;
                    nested
                  &lt;/div&gt;
                &lt;/div&gt;
              &lt;/Cell&gt;
            &lt;/Sheet&gt;
            &lt;Sheet name='hoooo'&gt;
              &lt;Cell row={1} col={2} cellStyle={{font: {bold: true}}}&gt;
                I am bored
              &lt;/Cell&gt;
              &lt;Cell cellRef='C9'&gt;
                plain string
              &lt;/Cell&gt;
            &lt;/Sheet&gt;
            &lt;XTrigger action='onClick'&gt;
              &lt;button&gt;
                get XLSX
              &lt;/button&gt;
            &lt;/XTrigger&gt;
          &lt;/WorkBook&gt;)
}
&nbsp;
const pusher = (x) =&gt; {
  let i = 0
  let a = []
  while (i &lt; x) {
    i++
    a.push(i)
  }
  return a
}
const sheets = pusher(9)
let cols = pusher(10)
let rows = pusher(10)
&nbsp;
export const big = () =&gt; {
&nbsp;
  return (&lt;WorkBook defaultCellStyle={defaultCellStyle} toXLSXCallback={handleXLSX} render={false}&gt;
            {sheets.map(sheet =&gt;
              &lt;Sheet key={sheet} name={'sheet ' + sheet}&gt;
                {cols.map(col =&gt;
                  rows.map(row =&gt;
                    &lt;div
                      row={row}
                      col={col}&gt;
                      {col + ', ' + row}
                    &lt;/div&gt;)
              )}
              &lt;/Sheet&gt;)}
&nbsp;
            &lt;XTrigger action='onClick'&gt;
              &lt;button&gt;
                get XLSX
              &lt;/button&gt;
            &lt;/XTrigger&gt;
          &lt;/WorkBook&gt;)
}
&nbsp;
export const subtest = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >  return (&lt;WorkBook defaultCellStyle={defaultCellStyle} toXLSXCallback={handleXLSX} render&gt;</span>
            &lt;SubTest cellRef='C9' content='feesh'/&gt;
            &lt;SubTest cellRef='C10' content='feesh'/&gt;
            &lt;XTrigger action='onClick'&gt;
              &lt;button&gt;
                get XLSX
              &lt;/button&gt;
            &lt;/XTrigger&gt;
          &lt;/WorkBook&gt;)
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Sat Sep 17 2016 11:09:47 GMT+0100 (GMT Daylight Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
