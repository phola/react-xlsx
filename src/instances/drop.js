import { workBookFromFile } from '../index'
import React, { PropTypes, Component } from 'react'
import Dropzone from 'react-dropzone'
export default class Drop extends React.Component {
  constructor () {
    super()
    this.state = { files: [] }
  }

  onDrop (files) {
    const updateState = file => {
      this.setState({
        files: [...this.state.files, ...[file]]
      })
    }
    files.forEach(file => {
      workBookFromFile(file, w => {
        file.Workbook = w
        updateState(file)
      })
    })
  }

  render () {
    return (
      <section>
        <div className='dropzone'>
          <Dropzone onDrop={this.onDrop.bind(this)}>
            <p>
              Try dropping some files here, or click to select files to upload.
            </p>
          </Dropzone>
        </div>
        <aside>
          <h2>Dropped files</h2>
          <ul>
            {this.state.files.map(f => (
              <li>
                {f.name} - {f.size} bytes
                <ul>
                  <h3>Sheets</h3>
                  {f.Workbook.SheetNames.map(s => (
                    <li>
                      {s}
                    </li>
                  ))}
                </ul>
              </li>
            ))}
          </ul>
        </aside>
      </section>
    )
  }
}
