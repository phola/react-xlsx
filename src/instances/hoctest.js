import React, { PropTypes, Component } from 'react'
import { WorkBookHOC, WorkBook, Sheet, Cell, XTrigger } from '../index'
import SubTest from './subtest'
import FileSaver from 'file-saver'

class HocTest extends Component {
  render () {
   return (<div>
            <Sheet name='hoooo'>
            <SubTest cellRef='C9' content='feesh'/>
            <SubTest cellRef='C10' content='feesh'/>
            </Sheet>
            <Sheet name='boooo'>
            <SubTest cellRef='C9' content='foossh'/>
            <SubTest cellRef='C10' content='foosh'/>
            </Sheet>
            </div>)
  }
}

const handleXLSX = (blob) => {
          console.log(blob)
          FileSaver.saveAs(blob, 'formulae.xlsx')
        }

export default WorkBookHOC(HocTest, {render: false, debug: false, generate: true, callback: handleXLSX})
