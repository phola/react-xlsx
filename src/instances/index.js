import React from 'react'
import { WorkBook, Sheet, Cell, XTrigger } from '../index'
import SubTest from './subtest'
import FileSaver from 'file-saver'

const defaultCellStyle = { font: { name: 'Verdana', sz: 11, color: 'FF00FF88' }, fill: { fgColor: { rgb: 'FFFFAA00' } } }

const handleXLSX = (blob) => {
  console.log(blob)
  FileSaver.saveAs(blob, 'test.xlsx')
}

export const simple = () => {
  return (<WorkBook defaultCellStyle={defaultCellStyle} toXLSXCallback={handleXLSX}>
            <Sheet name='woooo'>
              <Cell
                row={0}
                col={1}
                colSpan={2}
                rowSpan={6}>
                <div>
                  <div>
                    nested
                  </div>
                </div>
              </Cell>
            </Sheet>
            <Sheet name='hoooo'>
              <Cell row={1} col={2} cellStyle={{font: {bold: true}}}>
                I am bored
              </Cell>
              <Cell cellRef='C9'>
                plain string
              </Cell>
            </Sheet>
            <XTrigger action='onClick'>
              <button>
                get XLSX
              </button>
            </XTrigger>
          </WorkBook>)
}

const pusher = (x) => {
  let i = 0
  let a = []
  while (i < x) {
    i++
    a.push(i)
  }
  return a
}
const sheets = pusher(9)
let cols = pusher(10)
let rows = pusher(10)

export const big = () => {

  return (<WorkBook defaultCellStyle={defaultCellStyle} toXLSXCallback={handleXLSX} render={false}>
            {sheets.map(sheet =>
              <Sheet key={sheet} name={'sheet ' + sheet}>
                {cols.map(col =>
                  rows.map(row =>
                    <div
                      row={row}
                      col={col}>
                      {col + ', ' + row}
                    </div>)
              )}
              </Sheet>)}

            <XTrigger action='onClick'>
              <button>
                get XLSX
              </button>
            </XTrigger>
          </WorkBook>)
}

export const subtest = () => {
  return (<WorkBook defaultCellStyle={defaultCellStyle} toXLSXCallback={handleXLSX} render>
            <SubTest cellRef='C9' content='feesh'/>
            <SubTest cellRef='C10' content='feesh'/>
            <XTrigger action='onClick'>
              <button>
                get XLSX
              </button>
            </XTrigger>
          </WorkBook>)
}
