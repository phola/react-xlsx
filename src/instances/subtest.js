import React, { PropTypes, Component } from 'react'
import { Cell } from '../index'

export default class SubTest extends Component {
  render () {
  const {content, cellRef} = this.props
  return <Cell cellRef={cellRef}>
            {content}
          </Cell>
  }
}
