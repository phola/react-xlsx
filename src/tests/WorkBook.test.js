import React from 'react';
import { WorkBook } from '../index';
import { simple, big } from '../instances'
import renderer from 'react-test-renderer';

console.error = jest.genMockFn();

it('simple renders', () => {
  const jsx = simple()
  const component = renderer.create(
    jsx
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('throws prop type error if has no cells', () => {
  const jsx = simple()
  const component = renderer.create(
    <WorkBook><div>ts</div></WorkBook>
  );
let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // Assert Results

      expect(console.error).toBeCalled();
      expect(console.error.mock.calls.length).toBeGreaterThan(0);
      expect(console.error.mock.calls[0][0]).toContain('Cell');

});

it('big renders', () => {
  const jsx = big()
  const component = renderer.create(
    jsx
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
