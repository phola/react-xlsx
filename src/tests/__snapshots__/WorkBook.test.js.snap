exports[`test big renders 1`] = `null`;

exports[`test simple renders 1`] = `
<div>
  <div>
    <div>
      <div>
        nested
      </div>
    </div>
  </div>
  <div>
    <span>
      I am bored
    </span>
    <span>
      plain string
    </span>
  </div>
  <button
    onClick={[Function anonymous]}>
    get XLSX
  </button>
</div>
`;

exports[`test throws prop type error if has no cells 1`] = `
<div>
  <div>
    ts
  </div>
</div>
`;
